modules/adopter-registration-impl/src/main/java/org/opencastproject/adopter/statistic/ScheduledDataCollector.java

```diff
@@ -224,8 +232,19 @@ public void run() {
 
       if (adopter.allowsStatistics()) {
         try {
-          String statisticDataAsJson = collectStatisticData(adopter.getAdopterKey(), adopter.getStatisticKey());
-          sender.sendStatistics(statisticDataAsJson);
+          StatisticData statisticData = collectStatisticData(adopter.getAdopterKey(), adopter.getStatisticKey());
+          sender.sendStatistics(statisticData.jsonify());
+          if (null != tobiraEndpoint) {
+            JsonObject tobiraJson = tobiraEndpoint.getStats();
+            // This is null in the case that Tobira hasn't sent any stats yet.
+            // This could be due to Tobira not existing, or because we've just rebooted.
+            if (null != tobiraJson) {
+              JsonObject blob = new Gson().fromJson("{}", JsonElement.class).getAsJsonObject();
+              blob.addProperty("statistic_key", statisticData.getStatisticKey());
```

Comment:
Not a comment about your code but I just noticed the `snake_case`. And checking that other data that is sent, everything seems to be snake case. So then I will change Tobira to also send snake case!

Reply:
This is an (unfortunate) artifact of the Java<->Python stuff integration.  The sever's Python is written in proper (more or less) `snake_case`, but that's obviously not going to match Opencast's codebase :(